blockchain/chain.go

```diff
@@ -450,20 +450,18 @@ func (b *BlockChain) pruneStakeNodes() {
 		return
 	}
 
-	// Push the nodes to delete on a list. This will typically end up being
-	// a single node since pruning is currently done just before each new
-	// node is created.  However, that might be tuned later to only prune at
-	// intervals, so the code needs to account for the possibility of
-	// multiple nodes.
-	var deleteNodes []*blockNode
-	for node := pruneToNode.parent; node != nil; node = node.parent {
-		deleteNodes = append(deleteNodes, node)
-	}
-
-	// Loop through each node to prune in reverse, unlink its children, remove
-	// it from the dependency index, and remove it from the node index.
-	for i := len(deleteNodes) - 1; i >= 0; i-- {
-		node := deleteNodes[i]
+	// Add the nodes to prune to a slice.  This will typically end up being a
+	// single node since the pruning interval currently coincides with the
+	// average block time.
+	pruneNodes := make([]*blockNode, 0, 5)
```

Comment:
Just curious: was the "5" here just a guess?

Reply:
Yeah, I really just picked it based on intuition as the first number that popped into my head, and, given it's just a hint, if it isn't perfectly accurate, nothing bad will happen.

However, thinking about it a bit, I already know from previous work that ~99% of the blocks will be found within 25 minutes on mainnet and the target pruning time is 5 minutes, so 25/5=5.

It's a good point that I really used a lot of tribal knowledge for that intuition, so I'll update this PR to go ahead and calculate the value based on the network params to put in there so it's both accurate for all networks and isn't just a magic number seemingly out of nowhere.

Reply:
Updated to calculate the value when the pruner is created with appropriate comments to explain what is going on.  The calculated value does indeed end up matching my intuition for mainnet.

I went ahead and made this modify the pruning interval to match the target block time as well, which means the hint will match with 99% confidence on all networks.